spec/controler_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'controler'
+require 'user_interface'
+
+RSpec.describe Controler do
+  let(:actions) { [double('NullAction')] }
+  let(:output) { StringIO.new }
+  let(:valid_input) { StringIO.new('1') }
+
+  it 'tells the user_interface to print the menu' do
+    ui = UserInterface.new(valid_input, output)
+    ctrl = Controler.new(ui, actions)
+
+    allow(actions[0]).to receive(:run)
```

Comment:
you don't need to use a double in all tests, just for the test where you check that the right action was run :+1: 

Also, for the test where you check that the right action was run, instead of referencing `actions[0]`, you should reference the double directly:

```ruby
action_double = double('NullAction', run: nil)
controller = Controler.new(ui, [action_double])

controller.start

expect(action_double).to have_received(:run).once
```